packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -119,6 +119,14 @@ class AndroidWorkflow extends DoctorValidator implements Workflow {
 
     messages.add(new ValidationMessage('Android SDK at ${androidSdk.directory}'));
 
+    messages.add(new ValidationMessage(androidSdk.ndkDirectory == null
```

Comment:
Is this something we should report to users? Most won't need an NDK configured, and showing a message here would indicate to them that they need to do more setup.

Reply:
Going forward there might be more uses than just `--prefer-share-library`, e.g. profiling, ... and it's imho not a good user experience if these features sometimes work and sometimes not without telling the user if is setup is correct.

If I recall correctly, `flutter doctor` will not show it up with a red cross but in a more subtle way., so I think it makes sense to tell the user if the NDK was found or not.

Reply:
It currently shows up like this:

<img width="677" alt="screen shot 2017-11-30 at 1 01 55 pm" src="https://user-images.githubusercontent.com/1269969/33454772-b76a48b2-d5ce-11e7-9415-f38b34d27f00.png">

So:
- it shows as not being configured
- doesn't show as a warning about that (which is inconsistent with how we show other things)
- is something that most users will not need to configure, but given that doctor complains about it, users will think that they do need to do so

We may want to soften the warning and give the user some context about why and when they would want to configure. That could be something like only showing one line item for both the NDK + compiler, and using text like `Android NDK not configured; you may want to do this if...`.

Reply:
I have no strong opinion on it, except that there should be an indication if it was / was not discovered. Marking it as a red cross / warning is too much, since for now most people don't have to have it.

One line sounds good. Do you want me to make a change for it?

Reply:
> except that there should be an indication if it was / was not discovered

makes sense

> One line sounds good. Do you want me to make a change for it?

Sounds great, thanks!

Reply:
+1 to fixing this sooner than later. Users are already getting confused by it, see https://github.com/flutter/flutter/issues/13306.

Reply:
Will make a CL when I come in tomorrow!

Reply:
The pull request is up https://github.com/flutter/flutter/pull/13348

Reply:
Thank you!